### PR TITLE
Adding TRANSFER.md instructions for transferring to the REC

### LINKED ISSUE #23: Add detailed instructions for transferring crate ownership to R-E-C
We should add instructions to let users know how they can transfer projects to this organization. Specifically:
1. How to transfer the repo to the R-E-C org
2. How to add the appropriate crates.io owners to the existing crate
3. How to get re-added to the R-E-C if the owner wants to retain ownership/control of the crate as well

We're going over this now with the transfer of `usbd-midi` and I realized we don't really have this documented anywhere to point to easily.

### PR DESCRIPTION
**Pull request by ryan-summers**
Fixes #23 by adding a new TRANSFER.md file.

### CODE REVIEW

**Comment by eldruin** [comment]
Thank you for writing this!
There is one problem I have hit in the past with regard to publishing permissions, though:
> [In order to invite a team as an owner one must be a member of that team](https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner)

So the process is slightly more complicated:
- If the person wants to join the r-e-c, that should be done first and then the transfer process is as detailed here.
- If the person does not want to join the r-e-c, crates.io publishing permission must be granted to an individual of the org (they are probably in contact with or know about someone already. Otherwise they can open an issue. Failing that they can contact me, for example).
    - This person can then grant publishing permission to the r-e-c/all team

I did not know you can transfer a repository to an org you do not belong to, though. The [documentation](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository#about-repository-transfers) seems a bit contradictory to me (you can transfer to anybody by writing their name and they get an email vs. you need permission to create repositories in the target org)

**Comment by jamesmunns** [comment]
@eldruin and @ryan-summers, is there anything we can do to get @eldruin's concerns addressed, and get this merged?

**Comment by ryan-summers** [comment]
Nope, I was out on vacation the last two weeks, which is why I wasn't touching it. Thanks for the reminder though, I'll review the requested changes and get this updated shortly.

**Comment by ryan-summers** [comment]
Alright, this should be ready for another read-through now. Thanks for the review (+ ping)!

File: TRANSFER.md [hunk 1]
```diff
@@ -8,7 +8,17 @@ projects that they wish to transfer as well.
...
 
-1. Request to transfer the repository under the repository settings:
+The process for transferring ownership depends on whether or not the original crate owner would like
+to join the `rust-embedded-community` project.
+
+### Transfer Process (Joining the R-E-C)
+
+This process is used if the original crate owner would like to become a member of the
+`rust-embedded-community` project. This is required if the original owner still desires
+administration permissions for the source code repository of the crate.
```
Comment (eldruin) [inline TRANSFER.md:18]:
```suggestion
`rust-embedded-community` project.
```
We can still grant permissions to the repo.

File: TRANSFER.md [hunk 2]
```diff
@@ -21,5 +31,21 @@ projects that they wish to transfer as well.
...
+    1. Navigate to transfer the repository under the repository settings.
+    `https://github.com/<YOUR-ORG>/<YOUR-REPO>/settings`: General > Danger Zone > "Transfer
+    ownership"
+    2. Select "Specify an organization or username" and supply the Github username of one of the
+       `rust-embedded-community` members that you have been previously in contact with about
+       transferring the repository.
+        * If you haven't been in contact with anyone yet, feel free to [open an issue](https://github.com/rust-embedded-community/meta/issues/new) and one of us will be
+        happy to assist.
+    3. Confirm the transfer and select "I understand, transfer this repository"
+
```
Comment (eldruin) [inline TRANSFER.md:49]:
```suggestion
    4. Add the contact person to the crates.io owners to allow them to allow the community
     to issue updated releases for the crate:
   ```sh
   cargo owner --add github:<MEMBER-GITHUB-ID>
   ```

```

**Review by eldruin — COMMENTED** [review]
Thank you for coming back to this.
Just a couple of details left.

File: TRANSFER.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+       `rust-embedded-community` members that you have been previously in contact with about
+       transferring the repository.
+        * If you haven't been in contact with anyone yet, feel free to [open an issue](https://github.com/rust-embedded-community/meta/issues/new) and one of us will be
+        happy to assist.
+    3. Confirm the transfer and select "I understand, transfer this repository"
+    4. Add the contact person to the crates.io owners to allow them to allow the community
+     to issue updated releases for the crate:
+   ```sh
+   cargo owner --add github:<MEMBER-GITHUB-ID>
+Once the repository is transferred to the organizational member, that member should follow the steps
```
Comment (eldruin) [inline TRANSFER.md:52]:
```suggestion
   cargo owner --add <MEMBER-GITHUB-ID>
Once the repository is transferred to the organizational member, that member should follow the steps
```

**Review by eldruin — COMMENTED** [review]
Sorry, there is a formatting mistake where the `cargo owner` command block is not properly closed due to it being from a review, where github apparently does not support triple quotes.
Could you fix that manually?

**Comment by ryan-summers** [comment]
> Sorry, there is a formatting mistake where the `cargo owner` command block is not properly closed due to it being from a review, where github apparently does not support triple quotes. Could you fix that manually?

Should be done. Interestingly, Github still rendered it properly with the formatting mistake. Thanks for pointing this out though, should be good now. Feel free to manually update the PR and merge if you'd like as well

**Review by eldruin — APPROVED** [review]
Looks good to me! Thank you for following through with this.

**Pull request merged**